### DISCUSSION TITLE
why the need to rewrite the redis url in "Blog (with comments)" template app ?

### DISCUSSION THREAD

**Discussion by AC90** [vercel/next.js#51177] [Help] [answered]
### Summary

Hi - I'm just trying to get the "Blog (with comments)" template going and I've spent a while trying to figure out why I was having trouble connecting to an Upstash redis instance. The problems I had seemed to be because of the 'fixUrl' in Redis.ts which was rewriting the url from 

`rediss://`

to 

`rediss://:`

why the second colon ? why the need to do this ? 

I've removed the call to fixUrl and all seems to be working, but I don't understand why it was needed in the first place and whether I'm now unknowingly taking some insecure route ?

Any help / thoughts would be great
Alex

### Additional information

```js
function fixUrl(url: string) {
  if (!url) {
    return ''
  }
  if (url.startsWith('redis://') && !url.startsWith('redis://:')) {
    return url.replace('redis://', 'redis://:')
  }
  if (url.startsWith('rediss://') && !url.startsWith('rediss://:')) {
    return url.replace('rediss://', 'rediss://:')
  }
  return url
}
```

**Comment by devjiwonchoi** [comment, marked as answer]
Including `:` after `//` is to pass an empty user since Redis didn't support `user` until version 6.

> At the moment (up to and including v4), Redis doesn't support users and only provides authentication against a global password.

I guess it's a legacy.

## Reference
- https://stackoverflow.com/questions/44344628/how-to-create-a-redis-cloud-connection-url-with-an-auth-password
